src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -803,7 +803,7 @@ protected override ComputedInstanceFieldLayout ComputeInstanceFieldLayout(Metada
                 return ComputeExplicitFieldLayout(type, numInstanceFields);
             }
             else
-            if (type.IsEnum || MarshalUtils.IsBlittableType(type) || IsManagedSequentialType(type))
+            if (type.IsEnum || (type.IsSequentialLayout && !type.ContainsGCPointers))
```

Comment:
```suggestion
            if (type.IsSequentialLayout && !type.ContainsGCPointers)
```
Can it be simplified to this? Or is the `IsEnum` check important?

Reply:
C# enums are auto-layout, so I wanted to keep this check.
https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgCmA7ArgWwAQCieA3ngGJ4C+QA===

Reply:
Enums are single field, so both sequential and auto layout algorithms should produce exact same output. I would be curious if there are situations where this makes a difference for valid enums.

Reply:
Although, enum layout should be the same if sequential or auto, so removing this would probably be okay.

Reply:
Using sequential layout turned out to be the easiest way to enforce that enum types get the right alignment (with auto layout, 8-byte enums were getting the wrong alignment on x86)

Reply:
I am wondering whether it is a bug or a feature of the auto layout algorithm.

Reply:
I'll give it a shot to change this behavior and see what breaks.

Reply:
Looking at the code, it seems to be an explicit behavior choice to never require stricter alignment than `TARGET_POINTER_SIZE` except where we have a type that "requires 8-byte alignment" on arm32.
https://github.com/dotnet/runtime/blob/7a62468ffbb90008e08b30471bfa94b2d4c58726/src/coreclr/vm/methodtablebuilder.cpp#L8217-L8234

The reason that enums fields don't fall into this case is that our signature walking for layout (at least in the sequential case) normalizes enums to their underlying type.

Reply:
I've updated the managed type system to normalize enums to their underlying type for sequential and explicit layout and removed this case again. Let me know what you think!

Reply:
Nice! LGTM.